examples/docs_snippets/docs_snippets/integrations/dbt/tutorial/downstream_assets/assets.py

```diff
@@ -36,7 +36,7 @@ def jaffle_shop_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
 # start_downstream_asset
 @asset(
     compute_kind="python",
-    deps=get_asset_key_for_model([jaffle_shop_dbt_assets], "customers"),
```

Comment:
What's the error message you got here?

Reply:
just type checker - didn't try runtime

```
Argument of type "AssetKey" cannot be assigned to parameter "deps" of type "Iterable[CoercibleToAssetDep] | None" in function "asset"
  Type "AssetKey" is not assignable to type "Iterable[CoercibleToAssetDep] | None"
    "AssetKey" is incompatible with protocol "Iterable[CoercibleToAssetDep]"
      "__iter__" is not present
    "AssetKey" is not assignable to "None"
```

Reply:
runtime likely would have been  `Iteration is not allowed on `@record`s.` 
https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_record/__init__.py#L561
open to suggestions on improving that

Reply:
What if we override `__getitem__` with a specialized error message in `AssetKey`?

"You have attempted to iterate over an AssetKey. We disallowed this behavior because more often than not it is unintentional and therefore a bug. If this is intentional, change you code to ...."